core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/rename.kt

```diff
@@ -41,17 +139,47 @@ public fun <T, C> DataFrame<T>.rename(vararg cols: ColumnReference<C>): RenameCl
 @AccessApiOverload
 public fun <T, C> DataFrame<T>.rename(vararg cols: KProperty<C>): RenameClause<T, C> = rename { cols.toColumnSet() }
 
+/**
+ * @include [CommonRenameDocs]
+ * @include [SelectingColumns.ColumnNames] {@include [SetRenameOperationArg]}
+ * ### Examples:
+ * ```kotlin
+ * // Rename "col1" to "width" and "col2" to "length"
+ * df.rename("col1", "col2").into("width", "length")
+ *
+ * // Renames "arrival_date" and "passport-ID" columns to "camelCase"
+ * df.rename("arrival_date", "passport-ID").toCamelCase()
+ * ```
+ * @param [columns\] The [Columns Names][String] used to select the columns of this [DataFrame] to group.
+ */
 public fun <T> DataFrame<T>.rename(vararg cols: String): RenameClause<T, Any?> = rename { cols.toColumnSet() }
 
+/**
+ * An intermediate class used in the [rename] operation.
+ *
+ * This class itself does not perform any renaming — it is a transitional step
+ * before specifying how to rename the selected columns.
+ * It must be followed by one of the renaming methods
+ * to produce a new [DataFrame] with renamed columns.
+ *
+ * The resulting columns will keep their original values and positions
+ * in the [DataFrame], but their names will be changed.
+ *
+ * Use the following methods to perform the conversion:
+ * - [into(name)][RenameClause.into] — renames selected columns to the specified names.
+ * - [into { nameExpression }][RenameClause.into] — renames selected columns using a custom expression,
+ *   which takes the column and its path and returns a new name.
+ * - [toCamelCase()][RenameClause.toCamelCase] — renames all selected columns to `camelCase`.
+ *
+ * See [Grammar][RenameDocs.Grammar] for more details.
+ */
 @HasSchema(schemaArg = 0)
```

Comment:
Looks like we need a catalogue / design doc with all annotations, invented this year @koperagen ?
Could you create a simple md doc with them?